File: Java/quicksort/gemma2-9b-it/Gemma2_9b_it.java
import java.util.Random;
public class Gemma2_9b_it {

    public static void quickSort(int[] arr, int low, int high) {
        if (low < high) {
            // Partition the array
            int partitionIndex = partition(arr, low, high);

            // Recursively sort the sub-arrays before and after the partition index
            quickSort(arr, low, partitionIndex - 1);
            quickSort(arr, partitionIndex + 1, high);
        }
    }

    private static int partition(int[] arr, int low, int high) {
        // Choose the rightmost element as the pivot
        int pivot = arr[high];

        // Index of smaller element
        int i = (low - 1);

        for (int j = low; j < high; j++) {
            // If the current element is smaller than or equal to the pivot
            if (arr[j] <= pivot) {
                i++;

                // Swap arr[i] and arr[j]
                swap(arr, i, j);
            }
        }

        // Swap arr[i+1] and arr[high] (pivot)
        swap(arr, i + 1, high);

        // Return the partition index
        return i + 1;
    }

    private static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    public static void main(String[] args) {
        Random random = new Random();
        int size = 1000;
        int[] arr = new int[size];

        for (int i = 0; i < size; i++) {
            arr[i] = random.nextInt(10000) + 1;
        }

        // Ordenando com QuickSort
        quickSort(arr, 0, arr.length - 1);
    }
}

File: Python/quicksort/gemma2-9b-it/gemma2-9b-it.py
def quick_sort(arr):
    """
    Sorts a list using the Quick Sort algorithm.

    Args:
        arr: The list to be sorted.

    Returns:
        The sorted list.
    """
    if len(arr) < 2:
        return arr  # Base case: already sorted

    pivot = arr[0]  # Choose the first element as the pivot
    less = [i for i in arr[1:] if i <= pivot]
    greater = [i for i in arr[1:] if i > pivot]

    return quick_sort(less) + [pivot] + quick_sort(greater)

import random

def main():
    arr = [random.randint(0, 10000) for _ in range(1000)]
    quick_sort(arr)

if __name__ == "__main__":
    main()